flax/linen/module.py

```diff
@@ -498,14 +502,14 @@ def __call__(self, x):
   :meth:`compact` wrapper.
   """
 
-  if typing.TYPE_CHECKING:
-    def __init__(*args, **kwargs):
-      # this stub makes sure pytype accepts constructor arguments.
-      pass
-
-    def __call__(self, *args, **kwargs) -> Any:
-      # this stub allows pytype to accept Modules as Callables.
-      pass
+  # if typing.TYPE_CHECKING:
+  #   def __init__(*args, **kwargs):
+  #     # this stub makes sure pytype accepts constructor arguments.
+  #     pass
+  #
+  #   def __call__(self, *args, **kwargs) -> Any:
+  #     # this stub allows pytype to accept Modules as Callables.
+  #     pass
```

Comment:
I can reintroduce this or remove it, however this is a Liskov violation.  Why do you need Modules to be arbitrary callables?

Reply:
It's a bit lazy agreed, but common. In Pytorch Module you have:
```
forward: Callable[..., Any] = _forward_unimplemented
```
which amounts to the same.